### PR TITLE
Fix #26: Implement parameter recovery targets for method comparison

### PR DESCRIPTION
**Pull request by seabbs**
## Summary

This PR implements the **structural foundation** for comprehensive parameter recovery targets as specified in issue #26. This provides the framework for comparing different methods of handling double interval censored data, with detailed model validation to follow in subsequent issues.

### ⚠️ Status: Structural Implementation

This PR establishes the **infrastructure and framework** for method comparison. **Detailed model validation and testing will be addressed in follow-up issues** #41-44.

### Methods Implemented

✅ **Primary Censored Method** - Novel marginalisation approach framework  
✅ **Naive Method** - Baseline comparison structure  
✅ **Ward et al. Latent Variable Method** - Current best practice implementation framework  
✅ **Primary Censored with fitdistrplus** - MLE estimation structure  

### Key Structural Features

- **Unified Fitting Grid**: Single framework for comparing all methods across scenarios
- **Test Mode Support**: Rapid development and CI/CD validation infrastructure
- **Stan Model Integration**: Pre-compilation targets for efficient model reuse
- **Standardised Output**: Consistent format across all methods with parameter estimates, diagnostics, and runtime
- **Configuration System**: JSON-based bridge between Rmd parameters and targets execution
- **Error Handling**: Robust framework for handling edge cases and convergence failures

### Technical Implementation

- All methods support both gamma and lognormal distributions
- Test mode enables quick validation with reduced scenarios (2 instead of 152)
- Stan settings simplified: 2 chains, 1000 warmup/sampling iterations
- Proper separation of compilation time from runtime measurements
- Comprehensive linting and code style improvements

### Follow-up Issues for Model Validation

**Each method requires detailed review of implementation, priors, and testing:**

- **Issue #41**: Review primarycensored model implementation
- **Issue #42**: Review naive model implementation  
- **Issue #43**: Review Ward et al. latent variable model implementation
- **Issue #44**: Review primarycensored MLE implementation

Each issue will include:
- Implementation review and prior validation
- Simple parameter recovery tests inspired by primarycensored documentation
- Careful testing of edge cases and convergence
- Documentation of expected behaviour and limitations

### Test Plan Status

- [x] Stan models compile successfully
- [x] Test mode parameters work correctly  
- [x] Scenario subsetting functions properly
- [x] Data simulation pipeline generates expected outputs
- [x] Target structure renders without errors
- [x] Configuration system works robustly
- [x] Code linting and style consistency
- [ ] **Detailed model validation** (Issues #41-44)
- [ ] **Parameter recovery validation** (Issues #41-44)
- [ ] **Performance benchmarking** (Issues #41-44)

This PR provides the foundation; the follow-up issues will ensure each method is correctly implemented and validated.

Link to issue: Addresses #26

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

- **New Features**
  - Introduced comprehensive Bayesian and maximum likelihood model fitting workflows for censored delay distributions, supporting both simulation and Ebola case study data.
  - Added a unified fitting grid combining simulation and case study data with optional test mode for faster runs.
  - Provided utilities for data preparation, posterior extraction, and handling of distribution and truncation scenarios.
  - Added a GitHub Actions workflow for automated parameter recovery testing.
- **Bug Fixes**
  - Enhanced error handling in fitting functions to gracefully handle missing or invalid data.
- **Documentation**
  - Added detailed user documentation for all major functions and workflows.
- **Tests**
  - Included extensive unit tests covering fitting functions, data preparation, and edge cases.
- **Chores**
  - Updated package dependencies and exports to improve user access and development workflow.
  - Refined task runner commands and CI workflows for streamlined testing and coverage reporting.

### CODE REVIEW

File: _targets.Rmd [hunk 1]
```diff
@@ -164,6 +189,21 @@ grid$scenario_id <- paste(grid$distribution, grid$truncation, grid$censoring,
 grid$n <- simulation_n
 grid$seed <- seq_len(nrow(grid)) + base_seed
 
+# Subset for test mode if enabled
```
Comment (seabbs) [inline _targets.Rmd:192]: This is a messy place to do this. I think instead test mode should change the fitting fgrid

File: _targets.Rmd [hunk 2]
```diff
@@ -243,9 +283,12 @@ tar_target(
 We need Monte Carlo samples at different sizes for numerical validation.
 
 ```{targets sample_size_grid, tar_simple = TRUE}
+# Determine which sample sizes to use
```
Comment (seabbs) [inline _targets.Rmd:286]: Same comment here test mode should change the fitting grid
